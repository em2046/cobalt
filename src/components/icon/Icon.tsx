import ClassNameBuilder from '../../utils/class-name-builder'
import * as React from 'react'
import { IconType, iconSheet } from './icon-sheet-generated'

interface CustomIconProps {
  type: IconType
}

type NativeDivProps = React.HTMLAttributes<HTMLDivElement>
type IconProps = NativeDivProps & CustomIconProps

function isSpin(type: IconType) {
  return ['loading'].includes(type)
}

export default function Icon(props: IconProps) {
  const prefixClass = 'cobalt-icon'
  let classNameBuilder = new ClassNameBuilder(prefixClass, {
    [`${prefixClass}-spin`]: isSpin(props.type)
  })

  let classes = classNameBuilder.toString()

  let SVG = iconSheet[props.type]
  return (
    <span className={classes}>
      <SVG fill="currentColor" width="1em" height="1em" viewBox="0 0 32 32" />
    </span>
  )
}
