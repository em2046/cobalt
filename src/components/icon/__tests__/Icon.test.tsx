import React from 'react';
import renderer from 'react-test-renderer';
import Icon from '../Icon';

describe('Icon', () => {
  it('renders without crashing', () => {
    const component = renderer.create(<Icon type="close" />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
