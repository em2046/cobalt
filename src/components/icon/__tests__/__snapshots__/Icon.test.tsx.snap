// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon renders without crashing 1`] = `
<div
  className="cobalt-icon"
>
  <svg
    fill="currentColor"
    height="1em"
    viewBox="0 0 20 20"
    width="1em"
  >
    close.svg
  </svg>
</div>
`;
